packages/office-ui-fabric-react/src/utilities/responsiveMode/useResponsiveMode.tsx

```diff
@@ -0,0 +1,72 @@
+import * as React from 'react';
+import { getWindow } from '@uifabric/utilities';
+import { useOnEvent } from '@uifabric/react-hooks';
+import { ResponsiveMode } from './withResponsiveMode.types';
+
+const RESPONSIVE_MAX_CONSTRAINT = [479, 639, 1023, 1365, 1919, 99999999];
+
+/**
+ * User specified mode to default to, useful for server side rendering scenarios.
+ */
+let _defaultMode: ResponsiveMode | undefined;
+
+/**
+ * Tracking the last mode we successfully rendered, which allows us to
+ * paint initial renders with the correct size.
+ */
+let _lastMode: ResponsiveMode | undefined;
+
+const getResponsiveMode = (currentWindow: Window | undefined): ResponsiveMode => {
```

Comment:
Can any of this be shared with `withResponsiveMode` for now? The logic is complex enough that we should avoid duplicating it.